packages/store-engine-sqleet/src/index.ts

```diff
@@ -276,34 +248,36 @@ export class SQLeetEngine implements CRUDEngine {
       return record[RESERVED_COLUMN];
     }
 
-    return record;
+    return record as EntityType;
   }
 
   async readAll<T>(tableName: string): Promise<T[]> {
+    if (!this.db) {
+      throw new Error('Database is not instantiated');
+    }
     const table = this.schema[tableName];
     const columns = getFormattedColumnsFromTableName(table);
     const escapedTableName = escape(tableName);
 
     const selectRecordStatement = `SELECT ${columns} FROM ${escapedTableName};`;
-    let records = this.db.exec(selectRecordStatement);
-
-    // Ensure the record is not empty
-    if (records && records[0]) {
-      records = records[0];
-    } else {
-      return [];
-    }
+    const statement = await this.db.prepare(selectRecordStatement);
+    const records = (await statement.getAsObject()) as T[];
 
-    return records.values.map((record: any[]) => zipObject(records.columns, record));
+    return records;
   }
 
   async readAllPrimaryKeys(tableName: string): Promise<string[]> {
+    if (!this.db) {
```

Comment:
Please remove the ` if (!this.db)` checks. They are expectional. In most of the cases the database will be instantiated but then every db call makes this check. I guess the database will on `this.db.execute` already when there is no `this.db`, so we don't need to make the extra check for `this.db`.

Reply:
Thanks for pointing out. We used to have the this.db assignment in the init() therefore db was not set in the constructor (hence the extra ifs I added if we wanted to enable typescript types). Now, due to the changes in the websql API I was able to move the this.db assignment directly to the constructor (and remove the ifs).